packages/internal/bridge/sdk/src/tokenBridge.ts

```diff
@@ -547,143 +477,640 @@ export class TokenBridge {
       sender,
-      receipient,
-      req.amount,
-      req.token,
-      req.sourceChainId,
-      req.destinationChainId,
-      this.config.rootProvider,
-      req.gasMultiplier,
+      recipient,
+      amount,
+      token,
+      this.config.bridgeInstance.rootChainID,
     );
+    const txValue = (token.toUpperCase() === NATIVE)
+      ? amount.add(bridgeFee).toString() : bridgeFee.toString();
+    const unsignedBridgeTx : ethers.providers.TransactionRequest = {
+      data: txData,
+      to: this.config.bridgeContracts.rootERC20BridgeFlowRate,
+      value: txValue,
+      from: sender,
+      chainId: parseInt(this.config.bridgeInstance.rootChainID, 10),
+    };
+    sourceChainFee = this.calculateGasFee(feeData, rootGas.sourceChainGas);
+
+    const totalFees: ethers.BigNumber = sourceChainFee.add(approvalFee).add(bridgeFee).add(imtblFee);
+
+    return {
+      feeData: {
+        sourceChainGas: sourceChainFee,
+        approvalFee,
+        bridgeFee,
+        imtblFee,
+        totalFees,
+      },
+      contractToApprove,
+      unsignedApprovalTx,
+      unsignedBridgeTx,
+    };
+  }
+
+  private async getUnsignedBridgeWithdrawBundledTxPrivate(
+    sender: string,
+    recipient: string,
+    token: FungibleToken,
+    amount: ethers.BigNumber,
+    gasMultiplier: number,
+  ): Promise<BridgeBundledTxResponse> {
+    const [allowance, feeData, rootGas] = await Promise.all([
+      this.getAllowance(this.config.bridgeInstance.childChainID, token, sender),
+      this.config.childProvider.getFeeData(),
+      await this.getDynamicWithdrawGas(
+        this.config.bridgeInstance.rootChainID,
+        sender,
+        recipient,
+        token,
+        amount,
+      ),
+    ]);
+    // Get axelar fee
+    const axelarFee = await this.getAxelarFee(
+      this.config.bridgeInstance.childChainID,
+      this.config.bridgeInstance.rootChainID,
+      rootGas,
+      gasMultiplier,
+    );
+
+    let contractToApprove: string | null;
+    let unsignedApprovalTx: ethers.providers.TransactionRequest | null;
+    let sourceChainFee: ethers.BigNumber = ethers.BigNumber.from(0);
+    let approvalFee: ethers.BigNumber = ethers.BigNumber.from(0);
+    const bridgeFee: ethers.BigNumber = axelarFee;
+    const imtblFee: ethers.BigNumber = ethers.BigNumber.from(0);
+
+    // Approval required only for WIMX tokens with insufficient allowance.
+    if (this.isWrappedIMX(token, this.config.bridgeInstance.childChainID) && allowance.lt(amount)) {
+      contractToApprove = token;
+      const erc20Contract: ethers.Contract = await withBridgeError<ethers.Contract>(
+        async () => new ethers.Contract(token, ERC20, this.config.childProvider),
+        BridgeErrorType.PROVIDER_ERROR,
+      );
+      const data: string = await withBridgeError<string>(async () => erc20Contract.interface
+        .encodeFunctionData('approve', [
+          this.config.bridgeContracts.childERC20Bridge,
+          amount.sub(allowance),
```

Comment:
I don't believe this is correct. Approve would replace the [old allowance value not increment it](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L287)... so instead of the delta, I think we should be using the amount here?

Reply:
Yep,  this is a previous bug, just fixed.